src/features/PasswordAuth/ui/PasswordAuthModal.tsx

```diff
@@ -0,0 +1,23 @@
+import { Modal } from '@mantine/core'
+import { FC, useCallback } from 'react'
+import { PasswordAuth } from './PasswordAuth';
+
+interface Props {
+    opened?: boolean;
+    onClose?: () => void | undefined;
+}
+
+export const PasswordAuthModal: FC<Props> = ({ opened = false, onClose }) => {
+    const onModalClose = useCallback(() => {
+        onClose?.();
+    }, [onClose]);
+
+    return (
+        <Modal
+            opened={opened}
+            onClose={onModalClose}
+        >
+            <PasswordAuth />
```

Comment:
pass component from children 
```suggestion
            {children}
```